### PR TITLE
bpo-28415: Note 0 conversion different between Python and C

### PR DESCRIPTION
**Pull request by louisom**

### CODE REVIEW

**Review by zhangyangyu — CHANGES_REQUESTED** [review]
I think instead of a `.. note::`, it's better to follow the style of the table in https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting since we may find more exceptions in future. And I think the examples are not needed.

**Comment by serhiy-storchaka** [comment]
Too much words for minor difference.

**Comment by louisom** [comment]
@zhangyangyu @serhiy-storchaka After reference the description from [cpp - printf](http://www.cplusplus.com/reference/cstdio/printf/), simplify the note about it.

**Comment by zhangyangyu** [comment]
Actually I prefer the previous wording:
> the '0' conversion flag **IS NOT** ignored when a precision is given for 'd', 'i', 'o', 'u', 'x' and 'X' conversion types.

It's more direct to me and the current wording gives a feeling that the precision won't take effects at all. :-(
And I am not sure this deserves a highlight.

**Comment by terryjreedy** [comment]
See comment on https://bugs.python.org/issue28415 for minimal and terse alternatives.  Build error will disappear with either suggestion.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
See link to issue in comment

**Comment by rhettinger** [comment]
Just adding the two notes will suffice.  I don't think readers are helped at all by changing "exactly" to "nearly" over such a minor distinction.  The word "nearly" creates an unnecessary sense of risk and uncertainty in the minds of readers.   Also, I think this is so unimportant that it shouldn't be backported.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
The proposed addition is not proper English.  I give a different version on the issue.  I also think 'exactly' should just be removed when not exactly true, but will not push for that.

File: Doc/c-api/bytes.rst [hunk 1]
```diff
@@ -72,28 +72,28 @@ called with a non-bytes parameter.
    | :attr:`%c`        | int           | A single byte,                 |
    |                   |               | represented as a C int.        |
    +-------------------+---------------+--------------------------------+
-   | :attr:`%d`        | int           | Exactly equivalent to          |
+   | :attr:`%d`        | int           | Nearly equivalent to           |
    |                   |               | ``printf("%d")``.              |
```
Comment (vstinner) [inline Doc/c-api/bytes.rst]: Please add a reference to the note, like (1).

I would prefer to remove Nearly/Exactly. I don't think that the subtle difference is major enough to justify a "Nearly".

File: Doc/c-api/unicode.rst [hunk 2]
```diff
@@ -451,43 +451,43 @@ APIs:
    | :attr:`%c`        | int                 | A single character,            |
    |                   |                     | represented as a C int.        |
    +-------------------+---------------------+--------------------------------+
-   | :attr:`%d`        | int                 | Exactly equivalent to          |
+   | :attr:`%d`        | int                 | Nearly equivalent to           |
```
Comment (vstinner) [inline Doc/c-api/unicode.rst]: Ditto

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Doc/c-api/bytes.rst (hunk 1 above)
Comment (terryjreedy) [reply Doc/c-api/bytes.rst]: So we agree dropping 'exactly' when not exactly true.

**Comment by louisom** [comment]
@terryjreedy, @haypo, @zhangyangyu, could you help for review the latest addressed?
Thanks!

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
Replacing the original 'Exactly' with a note marker looks good to me.  I cannot apply patch to test generated html for correct formatting, but assuming you have, I approve net result.

**Comment by louisom** [comment]
I put down the unicode footnote under the original note block

**Comment by zhangyangyu** [comment]
Thanks @lulouie ! This style looks good and is actually what I mean at first.

**Pull request merged**